kotlinx-coroutines-core/common/src/flow/operators/Share.kt

```diff
@@ -303,8 +302,8 @@ public fun <T> Flow<T>.stateIn(
 ): StateFlow<T> {
     val config = configureSharing(1)
     val state = MutableStateFlow(initialValue)
-    scope.launchSharing(config.context, config.upstream, state, started, initialValue)
-    return state.asStateFlow()
+    val job = scope.launchSharing(config.context, config.upstream, state, started, initialValue)
+    return ReadonlyStateFlowWithJob(state, job)
```

Comment:
I would prefer having a single class with a nullable job in it just to keep things simple and concise (also, dex size) by the cost of 4 more bytes in a long-living object.

No strong opinion here though

Reply:
Done